src/Umbraco.Core/Runtime/MainDom.cs

```diff
@@ -83,7 +75,7 @@ public bool Register(Action release, int weight = 100)
         /// <returns>A value indicating whether it was possible to register.</returns>
         /// <remarks>If registering is successful, then the <paramref name="install"/> action
         /// is guaranteed to execute before the AppDomain releases the main domain status.</remarks>
-        public bool Register(Action install, Action release, int weight = 100)
+        public bool Register(Action install = null, Action release = null, int weight = 100)
```

Comment:
Should we throw if both `install` and `release` is null?

Reply:
I think it's still valid to call Register without anything since you might want to know if the result succeeded. Would be more or less the same as checking IsMainDom but isn't necessarily invalid to do